### PR TITLE
enable generic data import for rotki

### PR DESCRIPTION
**Pull request by prettyirrelevant**
Resolves backend aspect of #2270 

Currently, the import format chosen is...
`Transaction ID, Base Asset, Quote Asset, Type, Amount, Rate, Fee, Fee Currency, Description, Time`.

- `Amount` can be turned into `Buy Amount` & `Sell Amount` to improve ease of use for users. Thereby ruling out the need for a `Rate` column in the CSV template.
- `Transaction ID` can also be `ID`

This is a draft to know if the present approach is a step in the right direction.

### CODE REVIEW

**Review by LefterisJP — COMMENTED** [review]
So @prettyirrelevant what exactly do you want me to look at here?

The importer I see here seems to only be for trades.

```
Transaction ID,Base Asset,Quote Asset,Type,Amount,Rate,Fee,Fee Currency,Description,Time
092debd43022f98c02d1a7,USDC,ETH,Sell,1.0000,0.000625,,,Trade USDC for ETH,2022-07-29 09:00:00
```

But we also have all the other types of events such a transaction events, ledger actions, staking events etc.

These, along with the trades, are translated into `HistoryBaseEntry` seen [here](https://github.com/rotki/rotki/blob/develop/rotkehlchen/accounting/structures/base.py#L80).

Since trades are probably one of the most common structures it's okay to have our own trade format for import.

Now for the others though it woudl probably make sense to have them importable like that?


Also for the trade you can indeed have base amount and quote amount. I think it's better than rate.

What is the transaction id supposed to be though?

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            ),
+            notes=csv_row['Description'],
+        )
+        events.append(history_event)
+        if fee != ZERO and fee_currency is not None:
+            fee_event = HistoryBaseEntry(
+                event_identifier=HistoryBaseEntry.deserialize_event_identifier_from_kraken(identifier),  # noqa: E501
+                sequence_index=csv_row['sequence_index'] + 1,
+                timestamp=timestamp,
+                location=Location.EXTERNAL,
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_events.py:82]: Would it be interesting for the user to customize also the location? For example if I want to add some profits in a product of `location X` that we don't track?

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            ) * 1000,
+        )
+        fee = Fee(deserialize_asset_amount(csv_row['Fee'])) if csv_row['Fee'] else Fee(ZERO)  # noqa: E501
+        fee_currency = (
+            symbol_to_asset_or_token(csv_row['Fee Currency'])
+            if csv_row['Fee Currency'] and fee is not None else None
+        )
+        history_event = HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier_from_kraken(identifier),
+            sequence_index=csv_row['sequence_index'],
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_events.py:64]: If for some reason the user forgets to change the identifier or have it duplicated and the previous entry had a fee this will make two rows have the same `event_identifier` and `sequence_index` and the db insertion later will raise an error

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        """May raise:
+        - InputError if one of the rows is malformed
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for idx, row in enumerate(data):
+                try:
+                    # this is a requirement for history event entries
+                    row['sequence_index'] = idx
+                    self._consume_rotki_events(cursor, row, **kwargs)
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_events.py:105]: As for what I understood all the events are one line events. You could check that the identifier of the row wasn't seen before

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During Rotki generic CSV import, found action with unknown '
+                        f'asset {e.asset_name}. Ignoring entry',
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_events.py:109]: Add to the message the identifier of the row to detect it faster

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                    )
+                    continue
+                except DeserializationError as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'Deserialization error during Rotki generic CSV import. '
+                        f'{str(e)}. Ignoring entry',
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_events.py:115]: same, add the identifier of the row

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        May raise:
+        - UnsupportedCSVEntry if an unknown type is encountered.
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        try:
+            event_type, event_subtype = GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS[csv_row['Type']]  # noqa: E501
+        except KeyError as e:
+            raise UnsupportedCSVEntry(f'Unsupported entry {csv_row["Type"]}. Data: {csv_row}') from e  # noqa: E501
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_events.py:47]: Add the identifier of the row to detect it faster

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        )
+        amount_sold = deserialize_asset_amount(csv_row['Sell Amount'])
+        amount_bought = deserialize_asset_amount(csv_row['Buy Amount'])
+        trade = Trade(
+            timestamp=deserialize_timestamp_from_date(
+                date=csv_row['Timestamp'],
+                formatstr=timestamp_format,
+                location='Rotki generic trades importer',
+            ),
+            location=Location.EXTERNAL,
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_trades.py:45]: Same as before, should the location also be customizable?

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            fee=fee,
+            fee_currency=fee_currency,
+            rate=Price(amount_sold / amount_bought),
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_trades.py:48]: You should protect against a `DivisionByZero` error

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            base_asset=symbol_to_asset_or_token(csv_row['Base Currency']),
+            quote_asset=symbol_to_asset_or_token(csv_row['Quote Currency']),
+            trade_type=TradeType.SELL if csv_row['Type'] == 'Sell' else TradeType.BUY,
+            amount=amount_bought,
+            link=csv_row['ID'],
+            notes=csv_row.get('Description'),
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_trades.py:54]: Don't mix `.get` and `[]` for accessing dict entries . If the row doesn't have a note it value will be `None` already right?

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for row in data:
+                try:
+                    self._consume_rotki_trades(cursor, row, **kwargs)
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During Rotki generic CSV import, found action with unknown '
+                        f'asset {e.asset_name}. Ignoring entry',
```
Comment (yabirgb) [inline rotkehlchen/data_import/importers/rotki_trades.py:70]: As before also add the trade id to the error

File: rotkehlchen/tests/utils/dataimport.py [hunk 11]
```diff
@@ -1450,3 +1459,181 @@ def assert_binance_import_results(rotki: Rotkehlchen):
...
+            fee_currency=A_USD,
+            link='092debd43022f98c02d1a0',
+            notes='Trade ADA for BCH',
+        ),
+    ]
+    with rotki.data.db.conn.read_ctx() as cursor:
+        trades = rotki.data.db.get_trades(cursor, filter_query=TradesFilterQuery.make(), has_premium=True)  # noqa: E501
+        warnings = rotki.msg_aggregator.consume_warnings()
+    assert trades == expected_trades
+    assert len(warnings) == 0
```
Comment (yabirgb) [inline rotkehlchen/tests/utils/dataimport.py:1520]: Add some invalid rows also to verify that they fail as expected and are detected. Check that division by zero are detected for example

**Review by yabirgb — COMMENTED** [review]
Left you some comments. From the user side seeing the example csv seems okey. One of the thing that many find confusing (myself included) is the quote/base assets with the Type. Maybe would be interesting to consider having the columns as `sold asset`, `bought asset` and then in the trades always have sell/buy

**Comment by LefterisJP** [comment]
> Maybe would be interesting to consider having the columns as sold asset, bought asset and then in the trades always have sell/buy

No base / asset as names are not confusing and for this we would need to be consistent. Base/quote is standard in finance and trades and we should be consistent.

**Comment by yabirgb** [comment]
Also @prettyirrelevant about the identifier column in the trades https://github.com/rotki/rotki/blob/develop/rotkehlchen/exchanges/data_structures.py#L259 is created using the other fields. From the user side I believe it doesn't make much sense to have it there + you will get errors if you have duplicate identifiers

File: rotkehlchen/data_import/importers/rotki_events.py (hunk 6 above)
Comment (prettyirrelevant) [reply rotkehlchen/data_import/importers/rotki_events.py:47]: Since the entire row is returned here, is that still needed?

File: rotkehlchen/accounting/structures/base.py [hunk 12]
```diff
@@ -178,13 +178,14 @@ def serialized_event_identifier(self) -> str:
         return '0x' + hex_representation
 
     @classmethod
-    def deserialize_event_identifier_from_kraken(cls, val: str) -> bytes:
-        """Takes a `refid` from Kraken exchange and turns it into a bytes event_identifier."""
+    def deserialize_event_identifier_from_str(cls, val: str) -> bytes:
+        """Takes a `refid` from Kraken exchange or an arbitrary string identifier
+        and turns it into a bytes event_identifier."""
```
Comment (yabirgb) [inline rotkehlchen/accounting/structures/base.py:183]: The docstring was not updated

File: rotkehlchen/data_import/manager.py [hunk 13]
```diff
@@ -45,6 +49,10 @@ def get_importer_type(self) -> Type[BaseExchangeImporter]:
             return BisqTradesImporter
         if self == DataImportSource.BINANCE:
             return BinanceImporter
+        if self == DataImportSource.ROTKI_TRADES:
+            return RotkiGenericTradesImporter
+        if self == DataImportSource.ROTKI_EVENTS:
+            return RotkiGenericEventsImporter
```
Comment (yabirgb) [inline rotkehlchen/data_import/manager.py:55]: Add this new importer also to the api docs so the frontend can add them

File: rotkehlchen/tests/api/test_data_import.py [hunk 14]
```diff
@@ -453,6 +455,42 @@ def test_data_import_binance_history(rotkehlchen_api_server):
...
+
+    json_data = {'source': 'rotki events', 'file': str(filepath)}
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'dataimportresource',
+        ), json=json_data,
+    )
+    result = assert_proper_response_with_result(response)
+    assert result is True
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_data_import.py:490]: Why split it in two lines?

```suggestion
    assert assert_proper_response_with_result(response) is True
```

File: rotkehlchen/accounting/structures/base.py [hunk 15]
```diff
@@ -178,13 +178,14 @@ def serialized_event_identifier(self) -> str:
...
         return val.encode()
 
     @classmethod
-    def deserialize_event_identifier(cls, val: str) -> bytes:
-        """Takes an abritrary string and turns it into a bytes event_identifier.
+    def deserialize_event_identifier_from_tx_hash(cls, val: str) -> bytes:
+        """Takes a string tx_hash and turns it into a bytes event_identifier.
```
Comment (yabirgb) [inline rotkehlchen/accounting/structures/base.py:188]: Why have you also changed this one?

File: rotkehlchen/tests/data/rotki_generic_events.csv [hunk 16]
```diff
@@ -0,0 +1,8 @@
+ID,Type,Location,Currency,Amount,Fee,Fee Currency,Description,Timestamp
+1xyz,Deposit,kucoin,EUR,1000.00,,,Deposit EUR to Kucoin,2022-07-27 09:00:00
+2xyz,Withdrawal,binance,USDT,99.00,1.00,USDT,,2022-07-28 09:00:00
+3xyz,Withdrawal,kraken,BNB,1.01,,,,2022-07-29 09:00:00
+4xyz,Staking,luno,ETH,0.0513,,,ETH Staking reward from QRS,2022-08-01 09:00:00
+5xyz,Loss,coinbase,BTC,0.0910,,,,2022-08-02 09:00:00
+6xyz,Income,cex,DAI,1000.00,,,,2022-08-03 09:00:00
+7xyz,Invalid,bisq,BCH,0.3456,,,,2022-08-05 09:00:00
```
Comment (yabirgb) [inline rotkehlchen/tests/data/rotki_generic_events.csv:8]: From the user side how are these ids going to be handled? Are they needed or can be randomly created by us?

Comment (prettyirrelevant) [reply rotkehlchen/tests/data/rotki_generic_events.csv:8]: They're needed. If we randomly generate them, testing becomes difficult. 
Otherwise, randomly generating an event identifier works better.

File: rotkehlchen/accounting/structures/base.py (hunk 15 above)
Comment (prettyirrelevant) [reply rotkehlchen/accounting/structures/base.py:188]: Clarity sake

File: rotkehlchen/tests/data/rotki_generic_events.csv (hunk 16 above)
Comment (yabirgb) [reply rotkehlchen/tests/data/rotki_generic_events.csv:8]: Code and UX should come first. How is testing more complicated?

Comment (prettyirrelevant) [reply rotkehlchen/tests/data/rotki_generic_events.csv:8]: >  How is testing more complicated?

We'd be unable to know what the `event_identifier` is when comparing `expected_history_events` to `history_events` as we generate it randomly.
Unless I create an `assert_history_event_is_equal(history_event, expected_history_event)` function

File: rotkehlchen/accounting/structures/base.py [hunk 17]
```diff
@@ -169,7 +169,10 @@ def deserialize_from_db(cls, entry: HISTORY_EVENT_DB_TUPLE_READ) -> 'HistoryBase
     @property
     def serialized_event_identifier(self) -> str:
         """Take a HistoryBaseEntry's event_identifier and returns a string representation."""
-        if self.location == Location.KRAKEN:
+        # The only cases where this is used are in Kraken & Rotki generic events imports.
+        # For rotki events import ids, uuid4().hex is used which has a length of 32,
+        # hence the condition.
+        if self.location == Location.KRAKEN or len(self.event_identifier.decode()) == 32:
```
Comment (yabirgb) [inline rotkehlchen/accounting/structures/base.py:175]: This feels like something that can break in the future. What if instead you prepepend something like `rotki_imported` and  then you can do `self.event_identifier.startswith(b'rotki_imported')`

File: rotkehlchen/tests/utils/dataimport.py [hunk 18]
```diff
@@ -1450,3 +1459,200 @@ def assert_binance_import_results(rotki: Rotkehlchen):
...
+    generated randomly upon import."""
+    assert actual.identifier == expected.identifier
+    assert actual.sequence_index == expected.sequence_index
+    assert actual.timestamp == expected.timestamp
+    assert actual.location == expected.location
+    assert actual.asset == expected.asset
+    assert actual.event_type == expected.event_type
+    assert actual.event_subtype == expected.event_subtype
+    assert actual.balance == expected.balance
+    assert actual.notes == expected.notes
```
Comment (yabirgb) [inline rotkehlchen/tests/utils/dataimport.py:1658]: You can serialize both of them, pop the identifier and compare the dics

File: rotkehlchen/accounting/structures/base.py [hunk 19]
```diff
@@ -178,13 +181,13 @@ def serialized_event_identifier(self) -> str:
...
         return val.encode()
 
     @classmethod
-    def deserialize_event_identifier(cls, val: str) -> bytes:
-        """Takes an abritrary string and turns it into a bytes event_identifier.
+    def deserialize_event_identifier_from_tx_hash(cls, val: str) -> bytes:
+        """Takes a tx_hash and turns it into a bytes event_identifier.
         May raise:
         - DeserializationError if value is not valid transaction hash.
         """
```
Comment (yabirgb) [inline rotkehlchen/accounting/structures/base.py:193]: This name is confusing at least for me since we have `EVMTxHash`. But this is valid for any string, Why not restore to its original name?

File: rotkehlchen/accounting/structures/base.py [hunk 20]
```diff
@@ -169,22 +169,22 @@ def deserialize_from_db(cls, entry: HISTORY_EVENT_DB_TUPLE_READ) -> 'HistoryBase
...
-
-        hex_representation = self.event_identifier.hex()
-        if hex_representation.startswith('0x') is True:
-            return hex_representation
-        return '0x' + hex_representation
+        except UnicodeDecodeError:
+            hex_representation = self.event_identifier.hex()
+            if hex_representation.startswith('0x') is True:
+                return hex_representation
+            return '0x' + hex_representation
```
Comment (yabirgb) [inline rotkehlchen/accounting/structures/base.py:178]: This looks bad. You are relaying on a error to change the logic. This is bad. Why did you change this? If is an error that you found in a test you need to find why it happened and see how the logic should be, not relay on a error catch like this

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+
+
+class RotkiGenericTradesImporter(BaseExchangeImporter):
+    def _consume_rotki_trades(
+            self,
+            cursor: DBCursor,
+            csv_row: Dict[str, Any],
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> None:
+        """Consume Rotki generic trades import CSV file.
```
Comment (LefterisJP) [inline rotkehlchen/data_import/importers/rotki_trades.py:32]:
```suggestion
        """Consume rotki generic trades import CSV file.
```

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        - DvisionByZero if `amount_bought` is ZERO
```
Comment (LefterisJP) [inline rotkehlchen/data_import/importers/rotki_trades.py:37]:
```suggestion
        - DivisionByZero if `amount_bought` is ZERO
```

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+        fee_currency = (
+            symbol_to_asset_or_token(csv_row['Fee Currency'])
+            if csv_row['Fee Currency'] and fee is not None else None
+        )
+        amount_sold = deserialize_asset_amount(csv_row['Sell Amount'])
+        amount_bought = deserialize_asset_amount(csv_row['Buy Amount'])
+        try:
+            location = Location.deserialize(csv_row['Location'])
+        except DeserializationError:
+            location = Location.EXTERNAL
```
Comment (LefterisJP) [inline rotkehlchen/data_import/importers/rotki_trades.py:49]: same question as before for location

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 24]
```diff
@@ -0,0 +1,96 @@
...
+        - InputError if one of the rows is malformed
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for row in data:
+                try:
+                    self._consume_rotki_trades(cursor, row, **kwargs)
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During Rotki generic trades CSV import, found action with unknown '
```
Comment (LefterisJP) [inline rotkehlchen/data_import/importers/rotki_trades.py:79]:
```suggestion
                        f'During rotki generic trades CSV import, found action with unknown '
```

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 25]
```diff
@@ -0,0 +1,96 @@
...
+                        f'asset {e.asset_name}. Ignoring entry',
+                    )
+                    continue
+                except DeserializationError as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'Deserialization error during Rotki generic trades CSV import. '
```
Comment (LefterisJP) [inline rotkehlchen/data_import/importers/rotki_trades.py:85]:
```suggestion
                        f'Deserialization error during rotki generic trades CSV import. '
```

File: rotkehlchen/data_import/importers/rotki_trades.py [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+                        f'{str(e)}. Ignoring entry',
+                    )
+                    continue
+                except DivisionByZero:
+                    self.db.msg_aggregator.add_warning(
+                        f'During Rotki generic trades import, csv row {str(row)} has zero '
```
Comment (LefterisJP) [inline rotkehlchen/data_import/importers/rotki_trades.py:91]: Please try to keep rotki lowercase

```suggestion
                        f'During rotki generic trades import, csv row {str(row)} has zero '
```

File: rotkehlchen/db/upgrades/v32_v33.py [hunk 27]
```diff
@@ -456,9 +456,9 @@ def _force_bytes_for_tx_hashes(cursor: 'DBCursor') -> None:
         history_event = list(history_event)
         try:
             if is_valid_ethereum_tx_hash(history_event[1]):
-                history_event[1] = HistoryBaseEntry.deserialize_event_identifier(history_event[1])
+                history_event[1] = HistoryBaseEntry.deserialize_event_identifier_from_tx_hash(history_event[1])  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v32_v33.py:459]: The if check abstraction method should also be here

File: docs/api.rst [hunk 28]
```diff
@@ -10450,9 +10450,9 @@ Data imports
       Host: localhost:5042
       Content-Type: application/json;charset=UTF-8
 
-      {"source": "cointracking.info", "filepath": "/path/to/data/file", "timestamp_format": "%d/%m/%Y %H:%M:%S"}
+      {"source": "cointracking", "filepath": "/path/to/data/file", "timestamp_format": "%d/%m/%Y %H:%M:%S"}
```
[the remark this answers is not in the document]
Comment (prettyirrelevant) [reply docs/api.rst:10453]: `cointracking` is the one being used. A fix for this bug is already merged in bugfixes #4720

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+    ) -> str:
+        """Consume Rotki generic events import CSV file.
+        Returns the event identifier
+        May raise:
+        - UnsupportedCSVEntry if an unknown type is encountered.
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        identifier = f'rotki_events_{uuid4().hex}'
```
[the remark this answers is not in the document]
Comment (prettyirrelevant) [reply rotkehlchen/data_import/importers/rotki_events.py:47]: To ensure each history base entry is unique.

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 30]
```diff
@@ -0,0 +1,129 @@
...
+        try:
+            event_type, event_subtype = GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS[csv_row['Type']]  # noqa: E501
+        except KeyError as e:
+            raise UnsupportedCSVEntry(f'Unsupported entry {csv_row["Type"]}. Data: {csv_row}') from e  # noqa: E501
+        try:
+            location = Location.deserialize(csv_row['Location'])
+        except DeserializationError:
+            location = Location.EXTERNAL
```
[the remark this answers is not in the document]
Comment (prettyirrelevant) [reply rotkehlchen/data_import/importers/rotki_events.py:55]: The whole reason of the generic events import is to allow users to import data from any source provided it can be converted to our import format.

Because of this, we most definitely won't support some locations provided which is why `Location.EXTERNAL` is used in such instance.

File: rotkehlchen/accounting/structures/base.py [hunk 31]
```diff
@@ -169,26 +169,24 @@ def deserialize_from_db(cls, entry: HISTORY_EVENT_DB_TUPLE_READ) -> 'HistoryBase
...
             return hex_representation
         return '0x' + hex_representation
 
-    @classmethod
-    def deserialize_event_identifier_from_kraken(cls, val: str) -> bytes:
-        """Takes a `refid` from Kraken exchange and turns it into a bytes event_identifier."""
-        return val.encode()
-
     @classmethod
     def deserialize_event_identifier(cls, val: str) -> bytes:
```
Comment (LefterisJP) [inline rotkehlchen/accounting/structures/base.py:181]: I actually meant to keep the old functions and create the new combined one and use them where needed, but what you did is also fine though it uses one extra if in many cases it  can avoid it.

File: rotkehlchen/data_import/importers/rotki_events.py [hunk 32]
```diff
@@ -0,0 +1,129 @@
...
+                    usd_value=ZERO,
+                ),
+                notes=csv_row['Description'],
+            )
+            events.append(fee_event)
+        self.add_history_events(cursor, events)
+        return identifier
+
+    def _import_csv(self, cursor: DBCursor, filepath: Path, **kwargs: Any) -> None:
+        """May raise:
```
[the remark this answers is not in the document]
Comment (prettyirrelevant) [reply rotkehlchen/data_import/importers/rotki_events.py:89]: Took a while to understand this... but yeah that's a similar behaviour across all importers.

My thoughts? Failing only on `KeyError` shows a fundamental error i.e the CSV format is malformed.
In the case of other exceptions e.g `UnknownAsset`, `DeserializationError`, `UnsupportedCSVEntry`, etc. They are not so critical errors like `KeyError`.

Then again, maybe we should fail on all errors, idk?

[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/data_import/importers/rotki_events.py:89]: Yes you are correct. The CSV import would break from the very start in key error.
